core/transcoder.go

```diff
@@ -77,6 +152,79 @@ func (nv *NvidiaTranscoder) Transcode(job string, fname string, profiles []ffmpe
 	return resToTranscodeData(res, out)
 }
 
+func testOneNvidiaDevice(fname, gpu string, start chan interface{}, res chan error) {
+	<-start
+	t1 := NewNvidiaTranscoder(gpu)
+	// t1 := NewLocalTranscoder("/tmp")
+	profiles := []ffmpeg.VideoProfile{ffmpeg.P720p60fps16x9}
+	profiles[0].Format = ffmpeg.FormatMP4
+	// fname = "rendtest.mp4"
+	td, err := t1.Transcode("", fname, profiles)
+
+	t1.Stop()
+	if err != nil {
+		res <- err
+		return
+	}
+	if len(td.Segments) == 0 || td.Pixels == 0 {
+		res <- errors.New("Empty transcoded segment")
+		return
+	}
+	res <- nil
+}
+
+// TestNvidiaTranscoder tries to transcoded test segment on all the devices
+func TestNvidiaTranscoder(gpu string) error {
+	devices := strings.Split(gpu, ",")
+	b := bytes.NewReader(testSegment)
+	z, err := gzip.NewReader(b)
+	if err != nil {
+		return err
+	}
+	mp4testSeg, err := ioutil.ReadAll(z)
+	z.Close()
+	if err != nil {
+		return err
+	}
+	tmp, err := ioutil.TempDir("", "")
+	if err != nil {
+		return err
+	}
+	if true {
+		defer os.RemoveAll(tmp)
+	}
+	fname := path.Join(tmp, "testseg.mp4")
+	glog.Infof(fname)
+	err = ioutil.WriteFile(fname, mp4testSeg, 0644)
+	if err != nil {
+		return err
+	}
+	start := make(chan interface{})
+	res := make(chan error)
+	const sessionsToTest = 4
+	for _, device := range devices {
+		for i := 0; i < sessionsToTest; i++ {
```

Comment:
If the intent is to detect whether there's a session cap then a more straightforward way might be to transcode multiple renditions (the driver counts each rendition as one session). So, something like this:

```
p := ffmpeg.Profiles{Resolution:"16x16", Bitrate: "1k", Format: ffmpeg.FormatMP4}
_, err := tc.Transcode("", fname, []ffmpeg.Profiles{p, p, p, p})
```

I suspect this would make much of the concurrency around the `testOneNvidiaDevice` unnecessary.